### DISCUSSION TITLE
invalid format of client's host in testclient

### DISCUSSION THREAD

**Discussion by pjknkda** [encode/starlette#2109] [Potential Issue] [answered]
According to ASGI specification, `scope["client"]` for HTTP and Websocket protocol should have its host value in either IPv4 or IPv6 address formats.

However, in `starlette/testclient.py`, the `_TestClientTransport` set its host value to `"testclient"` and breaks applications that expect IPv4 or IPv6 address formats during tests.

https://github.com/encode/starlette/blob/f7bf74194358cceb5708f758989df89240c0c1aa/starlette/testclient.py#L246

https://github.com/encode/starlette/blob/f7bf74194358cceb5708f758989df89240c0c1aa/starlette/testclient.py#L264

**Comment by bevinhex** [comment]
Same here, it is breaking my tests that expecting IP

**Comment by toxadx** [comment]
The specification of ASGI message format: https://asgi.readthedocs.io/en/latest/specs/www.html?highlight=scope%5B%22client%22%5D#http-connection-scope

> client (Iterable[Unicode string, int]) – A two-item iterable of [host, port], where **host is the remote host’s IPv4 or IPv6 address**, and port is the remote port as an integer. Optional; if missing defaults to None.

**Comment by Kludex** [comment, marked as answer]
I've created #2295.
